xt/testapp/t/004_selenium.t

```diff
@@ -0,0 +1,268 @@
+
+The method used to find the "password_fn" on the IdP login page. The supported
+methods are: class, class_name, css, id, link, link_text, partial_link_text,
+tag_name, name, xpath.
+
+=item * L<login_btn>
+
+The identifier used to find the login button on th IdP's login page.  It is assumed
+that "clicking" the button will submit the username and/or password.
+
+Note that where there are separate username and password pages the same button
+identifier is assumed to be the same.
+
+=item * L<login_btn_by>
+
+The method used to find the "login_btn" on the IdP login page. The supported
+methods are: class, class_name, css, id, link, link_text, partial_link_text,
+tag_name, name, xpath.
+
+=item * L<post_login_page_title>
+
+Some IdPs provide a web page that is displayed after login and this allows the
+script to recoginize the page and move past it.
+
+=item * L<post_login_page_btn>
+
+The identifier used to find a button on the IdP's post login page.  It is assumed
+that "clicking" the button will move on to the NetSAML2 testapp's logged in page.
+
+=item * L<post_login_page_btn_by>
+
+The method used to find the "post_login_page_btn" on the IdP login page. The supported
+methods are: class, class_name, css, id, link, link_text, partial_link_text,
+tag_name, name, xpath.
+
+=item * L<post_logout_page_title>
+
+Some IdPs provide a web page that is displayed after logout and this allows the
+script to recoginize the page and move past it to the Net::SAML2 testapp login page.
+
+=item * L<login_bindings>
+
+Array of login bindings that the IdP supports (supported are: post, redirect).
+
+=item * L<logout_bindings>
+
+Array of logout bindings that the IdP supports (supported are: local, post, redirect).
+
+=item * L<issuer_value>
+
+Value of the issuer to find on the user logged in page for Net::SAML2 testapp to
+confirm that the login worked
+
+=back
+
+=cut
+
+my $driver =  Selenium::Chrome->new(
+                        driver => '~/perl-Net-SAML2/xt/testapp/t/selenium/chromedriver',
+                        accept_ssl_certs  => 1,
+                        );
+
+if ( ! $driver->get('https://netsaml2-testapp.local') ) {
+    die "Unable to access https://netsaml2-testapp.local";
+}
+
+$driver->pause(1000);
+
+##############################################
+# FIXME: Get this based on the list in IdPs
+##############################################
+my @idps = ("auth0");
+#my @idps = ("auth0", "azure", "google", "jump", "keycloak", "okta", "onelogin", "ping", "samltest", "shibboleth", "simplesamlphp");
+
+##############################################
+# Loop through each of the IdPs being tested
+##############################################
+foreach my $idp (@idps) {
+    print "==============================================\n";
+    print "= Beginning test for IdP: $idp\n";
+    print "==============================================\n";
+
+    # Load the IdP configuration file
+    my $config_file = 'IdPs/' . $idp . '/selenium.yml';
```

Comment:
I would use `File::Spec::Functions qw(catfile)` for this. Also makes paths Windows compatible.

Reply:
will do